backend/battling/views.py

```diff
@@ -1,74 +1,87 @@
 from django.shortcuts import redirect, render
+from django.views.generic.base import TemplateView
+from django.views.generic.edit import CreateView
 
 from .battles.battle import battle
 from .battles.email import send_battle_result
-from .forms import RoundForm, RoundForm2
+from .forms import CreatorForm, OpponentForm
 from .models import Battle
-from .pokemons.pokemon_data import check_valid_team1, check_valid_team2, get_pokemon_from_api
+from .pokemons.pokemon_data import get_pokemon_from_api, valid_team
 
 
-def home(request):
-    return render(request, "battling/home.html")
+class Home(TemplateView):
+    template_name = "home.html"
 
 
-def battle_new(request):
-    if request.method == "POST":
-        form = RoundForm(request.POST)
+class Invite(TemplateView):
+    template_name = "invite.html"
+
+
+class Opponent(TemplateView):  # will become an email
+    template_name = "opponent.html"
+
+
+class CreateBattle(CreateView):
+    model = Battle
+    form_class = CreatorForm
+    template_name = "create_battle.html"
+
+    def get_c(self, request):
```

Comment:
This should be the default signature of the get method from the `CreateView`;

`def get(self, request, *args, **kwargs):`